src/components/homepage/UpEvents.astro

```diff
@@ -0,0 +1,79 @@
+---
+import { loadConfig } from "../../utils/loadconfig";
+import { EventData } from "../../utils/types";
+
+const config = loadConfig();
+const upcoming_event_link = config.events.upcoming.link;
+
+let eventsData: EventData[];
+try {
+  const response = await fetch(upcoming_event_link);
+  if (!response.ok) {
+    throw new Error(`HTTP error! status: ${response.status}`);
+  }
+  eventsData = await response.json();
+  eventsData.forEach((event, index) => {
+    if (!event.BannerURL) {
+      event.BannerURL = "event.jpg";
```

Comment:
Adding the default image property configurable will be better idea. Since, some forks may wish to have their custom default image and its name may differ.